CODING_GUIDELINES.md

```diff
@@ -6,30 +6,86 @@ This document contains the coding guidelines for the ASPIRE Python project.
 
+The basic idea is that we have `feature` branches,  `develop` as an integration focused branch, and `master` for releases.
+We do not currently have a need for a dedicated release or staging branch; in our case `develop` also serves this purpose.
+Generally feature branches should branch off of the latest `develop` branch.
+
+Developers are welcome to maintain their own fork, but this is not required.
+
+When `develop` is at a milestone where we want to cut a release we will commit with a PEP440 compliant `Major.Minor` version
+scheme, tag it similarly, and complete a merge to master following the controls below. It is at this point the
+release would be a candadite to upload to PyPI.
+
+###  Git Workflow Controls
+
+To facilitate effective integration, merges into `develop` must occur as a pull request that is approved by two reviewers.
+One reviewer can be anyone, but should probably be someone related to the effort if that is applicable.
+The second developer is a designated codeowner described in the `CODEOWNERS` file.
+
+Additionally, all outstanding requested changes and comments must be resolved before merging.  Pushing new changes should trigger
+a re-review process of changed files to prevent oversights.
+
+Code should be passing integration and unit tests as defined in code on all supported plaforms by the evolving continuous integration systems,
+or be explicitly exempted by a codeowner. This is both for code quality and so maintenence of CI systems is not overlooked.
+
+All of these controls will be managed automatically by the Github server.
+
+(discussion note, maybe want an additional control (Amit?) on master/releases, I don't expect too often..).
+
+### Git Workflow Courtesy
+
+Contributors are responsible for keeping their working repositories up to date with the main Github repo.
+This is particularly important during active development.
+
+If a piece of work is known to change files that will effect other developers' active work, they should be invited to the review or tagged,
+so that the most effective order of integration can be established.  In the case of your own works having integration
+concerns, again, inviting other developers to discuss might be helpful.
+
+In the case a codeowner will not be available, such as vacation/travel, and an emergency patch can not be delayed
+a simple explicit consensus among the remaining developers should suffice.
+(discussion note, This level of urgency seems unlikely, can maybe drop and agree to delay, more likely a codeowner will just be busy,
+which is not as simple).
+
+Do not merge over another persons working branch or pull request unless explicitly permitted to do so by the author.
+Pull requests are welcomed.
+
+## Project Collaboration
+
+We will for a time attempt using Github Project boards in conjunction with Issues to document tasks in various stages.
```

Comment:
GitHub (not sure how “project board” should be capitalized, though)

Reply:
Fixed cap.

Regarding boards, the vendor does not, so I guess we shouldn't either. Changed

https://help.github.com/en/github/managing-your-work-on-github/about-project-boards

Reply:
I went ahead and added links to board and issues as well.